File: java/scripts/comandos/ban.java
// Command ban
// Exemplo: !ban <member>
// Linguagem usada: java
// Author: aquelemesmoojack

import java.time.temporal.ChronoUnit;

String[] args = event.getMessage().getContentRaw().split("\\s+");

if (args[0].equalsIgnoreCase(Main.prefix + "ban")) {
    if(args.length==1){
        event.reply("Você precisa especificar um membro para banir!").queue();
        return;
    }
    else if(event.getMentionedMembers().isEmpty()){
        event.getGuild().ban(args[1],0,"ban command").queue();
    }else{
        event.getGuild().ban(event.getMentionedMembers().get(0),0,"ban command").queue();
    }
    
}

File: lua/scripts/comandos/Ban.lua
--Ban command
--Exemplo: !ban <@member> <motivo>
--Linguagem usada: lua
--Author: aquelemesmoojack

function Ban:__init(data, parent)
    __init(self,data,parent)
    self.__init = self.client.__users:__insert(data.user)
end

function Ban:__hash()
    return self.__user.__id
end

function Ban:delete()
    return self.__parent:unbanUser(self.__user)
end

function get.guild(self)
    return self.__parent
end

function get.user(self)
    return self.__user
end

return Ban